creating a governance

miao0miao opened the pull request:
suggesting governance to define the roles in the community, their responsibilities, and how to navigate between them

miao0miao committed:
create governance

suggesting governance to define the roles in the community

miao0miao committed:
create governance

suggesting governance to define the roles in the community

tormath1 commented on governance:
```diff
@@ -0,0 +1,52 @@
+# overview
+
+
+Flatcra is a community based project, anyone who wants to prticipate is welcomed. We adopted the (CNCF code of conduct) [https://github.com/cncf/foundation/blob/main/code-of-conduct.md] as we pledge to be an opening and welcoming community for anyone who want to praticipate in it.
```
```suggestion
Flatcar is a community based project, anyone who wants to participate is welcomed. We adopted the (CNCF code of conduct) [https://github.com/cncf/foundation/blob/main/code-of-conduct.md] as we pledge to be an opening and welcoming community for anyone who want to participate in it.
```

ahrkrak replied:
```suggestion
Flatcar is a community based project, anyone who wants to participate is welcomed. We adopted the (CNCF code of conduct) [https://github.com/cncf/foundation/blob/main/code-of-conduct.md] as we pledge to be an opening and welcoming community for anyone who wants to participate in it.
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+
+- Collaborate well.
+- Demonstrate a deep and comprehensive understanding of Fltcar code base, technical goals, and directions.
+- Actively engage with major Chaos Mesh feature proposals and implementations.
```
```suggestion
- Actively engage with major Flatcar feature proposals and implementations.
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+
+### Maintainers
+
+Contribution per-se does not qulified to become a maintainer. Maintainers are first and foremost contributors that have shown they are committed to the long term success of a project. Contributors wanting to become maintainers are expected to be deeply involved in contributing code, pull request review, and triage of issues.
```
```suggestion
Contribution per se does not qualify someone to become a maintainer. Maintainers are first and foremost contributors that have shown they are committed to the long term success of a project. Contributors wanting to become maintainers are expected to be deeply involved in contributing code, pull request review, and triage of issues.
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+### Contributer
+
+
+Contributors are community members of who are engaged. Anyone can contribute to the project and become a contributor, regardless of their skillset. There is no expectation of commitment to the project, no specific skill requirements, and no selection process. There are many ways to contribute to the project, which may be one or more of the following (but not limited to):
```
```suggestion
Contributors are members of the Flatcar community who are engaged and wish to add to the project. Anyone can contribute to the project and become a contributor, regardless of their skillset. There is no expectation of commitment to the project, no specific skill requirements, and no selection process. There are many ways to contribute to the project, which may be one or more of the following (but not limited to):
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+
+- Reporting bugs and/or fixing them. 
+- posting feature requests
+- taking an ownership on maintineg spcific packages up to date. 
```
```suggestion
- Taking an ownership on maintineg spcific packages up to date. 
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+Contributors are community members of who are engaged. Anyone can contribute to the project and become a contributor, regardless of their skillset. There is no expectation of commitment to the project, no specific skill requirements, and no selection process. There are many ways to contribute to the project, which may be one or more of the following (but not limited to):
+
+- Reporting bugs and/or fixing them. 
+- posting feature requests
```
```suggestion
- Posting feature requests
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+- Reporting bugs and/or fixing them. 
+- posting feature requests
+- taking an ownership on maintineg spcific packages up to date. 
+- updating/writing documentation
```
```suggestion
- Updating/writing documentation
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+- posting feature requests
+- taking an ownership on maintineg spcific packages up to date. 
+- updating/writing documentation
+- submitting pull requests or patches
```
```suggestion
- Submitting pull requests or patches
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+- Joining discussions on Twitter, our community [Slack channel](https://kubernetes.slack.com/archives/C03GQ8B5XNJ) ,[Matrix channel](https://app.element.io/#/room/#flatcar:matrix.org) and office hours. 
+- Evangelizing about the project. 
+
+For first-time contributors, it’s recommended to start by going through [Contribuiting-git](https://github.com/flatcar-linux/Flatcar/blob/main/contributions-git.md), and joining our community Slack channel or Matrix channel. 
```
```suggestion
For first-time contributors, it’s recommended to start by going through [Contributing-git](https://github.com/flatcar-linux/Flatcar/blob/main/contributions-git.md), and joining our community Slack channel or Matrix channel. 
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+Committers are active community members who are responsible for the overall quality and stewardship of the project. Committers are part of the organization with write access to all repositories. Committers are expected to remain actively involved in the project and participate in voting and discussing of proposed project level changes.
+
+Typically, a potential committer needs to show that they have a sufficient understanding of the project, its objectives, and its strategy. To become a committer, you are expected to:
+- express an intresset to the exsiting maintainers about becoming cecoming a coomiter.
```
```suggestion
- express an interest to the existing maintainers about becoming becoming a committer; and
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+
+Typically, a potential committer needs to show that they have a sufficient understanding of the project, its objectives, and its strategy. To become a committer, you are expected to:
+- express an intresset to the exsiting maintainers about becoming cecoming a coomiter.
+- Have creted substentials PRs
```
```suggestion
- have created substantial PRs.
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+Committers wanting to become maintainers are expected to:
+
+- Collaborate well.
+- Demonstrate a deep and comprehensive understanding of Fltcar code base, technical goals, and directions.
```
```suggestion
- Demonstrate a deep and comprehensive understanding of the Flatcar code base, technical goals, and directions.
```

ahrkrak commented on governance:
```diff
@@ -0,0 +1,52 @@
+- Demonstrate a deep and comprehensive understanding of Fltcar code base, technical goals, and directions.
+- Actively engage with major Chaos Mesh feature proposals and implementations.
+
+Periodically, the existing maintainers curate a list of contributors that have shown regular activity on the project over the prior months. The nominating maintainer will create a PR to update the Maintainers List. It is recommended to describe the reasons for the nomination and the contribution of the nominee in the PR. Upon consensus of incumbent maintainers, the PR will be approved and the new ...
```
delete this para? I don't think we do this.

miao0miao replied:
I want to also lay the foundation for growth and establish a clear path for becoming a maintainer.

miao0miao replied:
@ahrkrak I think we should keep it, any objection?

miao0miao committed:
Apply suggestions from code review to the governance file

Co-authored-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
Co-authored-by: Andrew Randall <andrew.randall@gmail.com>

miao0miao committed:
Apply suggestions from code review to the governance file

Co-authored-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
Co-authored-by: Andrew Randall <andrew.randall@gmail.com>

miao0miao committed:
updating the code of conduct

miao0miao committed:
updating the code of conduct

miao0miao committed:
Rename governance to governance.md

miao0miao committed:
Rename governance to governance.md

miao0miao committed:
readme.md governance updated

miao0miao committed:
readme.md governance updated

miao0miao committed:
changing the code of conduct to MS open source

miao0miao committed:
changing the code of conduct to MS open source

ahrkrak commented on governance.md:
```diff
@@ -0,0 +1,52 @@
+- Joining discussions on Twitter, our community [Slack channel](https://kubernetes.slack.com/archives/C03GQ8B5XNJ) ,[Matrix channel](https://app.element.io/#/room/#flatcar:matrix.org) and office hours. 
+- Evangelizing about the project. 
+
+For first-time contributors, it’s recommended to start by going through [Contributing-git](https://github.com/flatcar-linux/Flatcar/blob/main/contributions-git.md), and joining our community Slack channel or Matrix channel. 
```
```suggestion
For first-time contributors, it’s recommended to start by going through [Contributing-git](https://github.com/flatcar/Flatcar/blob/main/contributions-git.md), and joining our community Slack channel or Matrix channel. 
```

miao0miao committed:
Andy - changing the Flatcar domain in a link

Co-authored-by: Andrew Randall <andrew.randall@gmail.com>

miao0miao committed:
Andy - changing the Flatcar domain in a link

Co-authored-by: Andrew Randall <andrew.randall@gmail.com>

t-lo committed:
README, governance: limit scope

This change limits the scope of governance to what is currently lived in
the project.

Signed-off-by: Thilo Fromm <thilo@kinvolk.io>

t-lo committed:
Merge branch 'miao0miao-patch-1-governance' into t-lo/governance-limit-scope

t-lo committed:
Merge pull request #863 from flatcar/t-lo/governance-limit-scope

governance: limit scope
